lib/pure/sugar.nim

```diff
@@ -131,6 +122,27 @@ macro `=>`*(p, b: untyped): untyped =
                    pragmas = pragma, name = name,
                    procType = kind)
 
+macro `=>`*(p, b: untyped): untyped =
+  ## Syntax sugar for anonymous procedures. It also supports pragmas.
+  runnableExamples:
+    proc passTwoAndTwo(f: (int, int) -> int): int = f(2, 2)
+
+    assert passTwoAndTwo((x, y) => x + y) == 4
+
+    type
+      Bot = object
+        call: (string {.noSideEffect.} -> string)
+
+    var myBot = Bot()
+
+    myBot.call = (name: string) {.noSideEffect.} => "Hello " & name & ", I'm a bot."
```

Comment:
Could we change the example here to another pragma (maybe `{.inline.}`)? The reason is that this is no longer the best practice since we have `|=>` so it's misleading to have it as an example.